lib/common/xml.c

```diff
@@ -3290,6 +3298,17 @@ dump_xml_unformatted(xmlNode * an_xml_node)
     return buffer;
 }
 
+void
+pcmk__xml_serialize_fd_formatted(int fd, xmlNode *cur)
+{
+    xmlOutputBuffer *fd_out = xmlOutputBufferCreateFd(fd, NULL);
+    CRM_ASSERT(fd_out != NULL);
+    xmlNodeDumpOutput(fd_out, cur->doc, cur, 0, 1, NULL);
+    xmlOutputBufferWrite(fd_out, sizeof("\n") - 1, "\n");  /* final NL */
+    xmlOutputBufferClose(fd_out);
+    fdatasync(fd);
```

Comment:
Apparently fdatasync() can be buggy in FreeBSD (and isn't available before FreeBSD 11.1). Since meta-data has to be sync'd due to file size change, there won't be any benefit over fsync() anyway.